jetstream/kv.go

```diff
@@ -484,6 +484,63 @@ func (js *jetStream) CreateKeyValue(ctx context.Context, cfg KeyValueConfig) (Ke
 		return nil, err
 	}
 
+// KeysWithFilters returns a filtered list of keys in the bucket.
+func (js *jetStream) KeysWithFilters(filter []string) ([]string, error) {
```

Comment:
Filtering the keys does not mean filtering them on client side code.
It means passing `subject filters` to `ordered consumer`. Take a look how `ListKeys` or `Keys` are implemented.